pkg/api/validation/validation.go

```diff
@@ -2083,9 +2155,51 @@ func ValidateReadOnlyPersistentDisks(volumes []api.Volume, fldPath *field.Path)
 	return allErrs
 }
 
+// validateTaints tests if given taints have valid data.
+func validateTaints(taints []api.Taint, fldPath *field.Path) field.ErrorList {
+	allErrors := field.ErrorList{}
+	for i, currTaint := range taints {
+		idxPath := fldPath.Index(i)
+		// validate the taint key
+		allErrors = append(allErrors, unversionedvalidation.ValidateLabelName(currTaint.Key, idxPath.Child("key"))...)
+		// validate the taint value
+		if !validation.IsValidLabelValue(currTaint.Value) {
```

Comment:
The regexp in `IsValidLabelValue()` is `^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$` which accepts empty string (the top level `( )?`means it) .